File: Coupons.js
import React from 'react';
import { startCoupons, stopCoupons } from './Api';

function App() {
    const handleStart = async () => {
        try {
            const data = await startCoupons();
            console.log(data.message);
        } catch (error) {
            console.error('Failed to start coupons:', error);
        }
    };

    const handleStop = async () => {
        try {
            const data = await stopCoupons();
            console.log(data.message);
        } catch (error) {
            console.error('Failed to stop coupons:', error);
        }
    };

    return (
        <div>
            <h1>Safeway Coupons Manager</h1>
            <button onClick={handleStart}>Start Coupons</button>
            <button onClick={handleStop}>Stop Coupons</button>
        </div>
    );
}

export default App;

File: coupons.py
from flask import Flask, jsonify
import subprocess

app = Flask(__name__)

@app.route('/start-coupons', methods=['POST'])
def start_coupons():
    try:
        # Start the Docker container
        subprocess.run(['docker-compose', 'up', '-d'], check=True)
        return jsonify({'message': 'Safeway Coupons process started successfully!'}), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@app.route('/stop-coupons', methods=['POST'])
def stop_coupons():
    try:
        # Stop the Docker container
        subprocess.run(['docker-compose', 'down'], check=True)
        return jsonify({'message': 'Safeway Coupons process stopped successfully!'}), 200
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    app.run(debug=True)